src/main/java/si/mazi/rescu/clients/ApacheConnection.java

```diff
@@ -0,0 +1,228 @@
+        return new ApacheConnection(url, proxy);
+    }
+    
+    private void exec() {
+        if (executed) {
+            return;
+        }
+        HttpClientBuilder clientBuilder = builder
+                .setDefaultRequestConfig(requestConfig.build());
+        if (proxy != null && proxy != Proxy.NO_PROXY) {
+            InetSocketAddress address = (InetSocketAddress) proxy.address(); 
+            HttpHost proxy = new HttpHost(address.getHostName(), address.getPort());
+            clientBuilder.setProxy(proxy);
+        }
+        
+        CloseableHttpClient client = clientBuilder
+                .build();
+        
+        HttpRequestBase request = createRequest(method, url);
+        
+        if (request instanceof HttpEntityEnclosingRequestBase && out != null) {
+            HttpEntityEnclosingRequestBase req = (HttpEntityEnclosingRequestBase) request;
+            HttpEntity entity = req.getEntity();
+            if (entity == null) {
+                entity = new ByteArrayEntity(out.toByteArray());
+                req.setEntity(entity);
+            }
+        }
+        headers.forEach(request::addHeader);
+        try {
+            response = client.execute(request);
+            
+        } catch (Throwable e) {
+            throw new RuntimeException(e);
+        }
+        executed = true;
+    }
+
+    private static HttpRequestBase createRequest(HttpMethod method, String url) {
+        switch(method) {
+            case GET: return new HttpGet(url); 
+            case POST: return new HttpPost(url);
+            case PUT: return new HttpPut(url);
+            case DELETE: return new HttpDelete(url);
+            case HEAD: return new HttpHead(url); 
+            case OPTIONS: return new HttpOptions(url);
+            case PATCH: return new HttpPatch(url);
+            default: throw new RuntimeException("Not supported http method: " + method);
+        }
+    }
+
+    @Override
+    public String getHeaderField(String name) {
+        try {
+            return res.getFirstHeader(name).getValue();
+        } catch (Exception e) {
+            return null;
+        }
+    }
+    
+    @Override
+    public void setRequestMethod(HttpMethod method) throws ProtocolException {
+        this.method = method;
+    }
+
+    @Override
+    public String getRequestMethod() {
+        return method.name();
+    }
+
+    @Override
+    public OutputStream getOutputStream() throws IOException {
+        if(out == null) {
+            out = new ByteArrayOutputStream();
+        }
+        return out;
+    }
+    
+    @Override
+    public InputStream getInputStream() throws IOException {
+        return response.getEntity().getContent();
+    }
+    
+    @Override
+    public InputStream getErrorStream() throws IOException {
+        return getInputStream();
+    }
+    
+    @Override
+    public void doFinalConfig(int contentLength) {
+        // TODO ???
```

Comment:
Remove todo, nothing to do here (empty method is ok). Can also be moved to the interface as the default implementation.